src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -711,6 +750,8 @@ public NodeViewModel(WorkspaceViewModel workspaceViewModel, NodeModel logic)
             DynamoViewModel.Model.PropertyChanged += Model_PropertyChanged;
             DynamoViewModel.Model.DebugSettings.PropertyChanged += DebugSettings_PropertyChanged;
 
+            WorkspaceViewModel.ZoomChanged += UpdateOverlays;
```

Comment:
I wonder if this should be throttled, have you tested with hundreds of nodes? How is performance?